src/blinding.cpp

```diff
@@ -298,6 +301,37 @@ std::pair<uc32, cleared_uc32> blind25_key_pair(
     return result;
 }
 
+static const auto version_blinding_hash_key_sig = to_unsigned_sv("VersionCheckKey_sig"sv);
+
+std::pair<uc32, cleared_uc64> blind_version_key_pair(ustring_view ed25519_sk) {
+    std::array<unsigned char, 64> ed_sk_tmp;
+    if (ed25519_sk.size() == 32) {
+        std::array<unsigned char, 32> pk_ignore;
+        crypto_sign_ed25519_seed_keypair(pk_ignore.data(), ed_sk_tmp.data(), ed25519_sk.data());
+        ed25519_sk = {ed_sk_tmp.data(), 64};
+    }
+    if (ed25519_sk.size() != 64)
+        throw std::invalid_argument{
+                "blind_version_key_pair: Invalid ed25519_sk is not the expected 32- or 64-byte "
+                "value"};
```

Comment:
Most of this is not needed: we really only just need a check for size == 32 or size == 64 (technically just >= 32 is enough, but it's probably worth erroring if we get some oddball size).    The reason it isn't needed is that `crypto_sign_ed25519_seed_keypair` doesn't do anything to the seed: it just copies it into the first 32-bytes of the result, then follows it by the 32-byte pubkey, but since we only use the 32-byte seed, this isn't something we care about and can just use the first 32 bytes of `ed25519_sk` as the seed.

Reply:
That's a bit confusing.  The point is:

`crypto_sign_ed25519_seed_keypair([seed32])` returns `[seed32, pk32]` packed into a 64-byte location, and so because we are only using the first 32 bytes anyway, we don't need the `crypto_sign_ed25519_seed_keypair` call (which only really produces new data in the pk32 that we are ignoring).

Reply:
Ah yep, nice - I'll remove it